### PR TITLE
add support for `-nographics` parameter

### PR DESCRIPTION
**Pull request by maxomatic458**
this has **NOT** been tested for 3.9.3 yet.

also see https://github.com/project-fika/Fika-Plugin/pull/104

### CODE REVIEW

File: Fika.Dedicated/Patches/TextureValidateFormatPatch2.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                }
+            }
+
+            return null;
+        }
+
+        [PatchPostfix]
+        static void Postfix(ref bool __result)
+        {
+            Logger.LogError($"ValidateFormat: {__result}");
```
Comment (ArchangelWTF) [inline Fika.Dedicated/Patches/TextureValidateFormatPatch2.cs:28]: Remove or log as debug

File: Fika.Dedicated/Patches/TextureValidateFormatPatch3.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                }
+            }
+
+            return null;
+        }
+
+        [PatchPostfix]
+        static void Postfix(ref bool __result)
+        {
+            Logger.LogError($"ValidateFormat: {__result}");
```
Comment (ArchangelWTF) [inline Fika.Dedicated/Patches/TextureValidateFormatPatch3.cs:28]: Remove or log as debug

**Review by ArchangelWTF — COMMENTED** [review]
Tested and seems to completely remove the game being held up by -nographics

LGTM apart from the small nits of debug logging ending up in the headless console.

File: Fika.Dedicated/Patches/TextureValidateFormatPatch2.cs (hunk 1 above)
Comment (maxomatic458) [reply Fika.Dedicated/Patches/TextureValidateFormatPatch2.cs:28]: forgot to remove those

File: Fika.Dedicated/Patches/TextureValidateFormatPatch1.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+using SPT.Reflection.Patching;
+using UnityEngine;
+using UnityEngine.Experimental.Rendering;
+
+namespace Fika.Dedicated.Patches {
+    // https://github.com/Unity-Technologies/UnityCsReference/blob/77b37cd9f002e27b45be07d6e3667ee53985ec82/Runtime/Export/Graphics/Texture.cs#L696
+    public class ValidateFormatPatch1 : ModulePatch {
+        protected override MethodBase GetTargetMethod()
+        {
+            var methods = typeof(Texture).GetMethods(BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (Lacyway) [inline Fika.Dedicated/Patches/TextureValidateFormatPatch1.cs:11]: use explicit vars

**Review by Lacyway — APPROVED** [review]
Works good so far from my testing, will merge and keep playing with it. Nice find!

**Pull request merged**